src/sagas/wallet.js

```diff
@@ -0,0 +1,541 @@
+      }
+    }
+  }
+
+  yield put(tokenMetadataUpdated(tokenMetadatas, errors));
+}
+
+// This will create a channel to listen for featureFlag updates
+export function* listenForFeatureFlags(featureFlags) {
+  const channel = eventChannel((emitter) => {
+    const listener = (state) => emitter(state);
+    featureFlags.on('wallet-service-enabled', (state) => {
+      emitter(state);
+    });
+
+    // Cleanup when the channel is closed
+    return () => {
+      featureFlags.removeListener('wallet-service-enabled', listener);
+    };
+  });
+
+  try {
+    while (true) {
+      const newUseWalletService = yield take(channel);
+      const oldUseWalletService = yield select((state) => state.useWalletService);
+
+      if (oldUseWalletService && oldUseWalletService !== newUseWalletService) {
+        walletHelpers.reloadElectron();
+      }
+    }
+  } finally {
+    if (yield cancelled()) {
+      // When we close the channel, it will remove the event listener
+      channel.close();
+    }
+  }
+}
+
+// This will create a channel from an EventEmitter to wait until the wallet is loaded,
+// dispatching actions
+export function* listenForWalletReady(wallet) {
+  const channel = eventChannel((emitter) => {
+    const listener = (state) => emitter(state);
+    wallet.on('state', (state) => emitter(state));
+
+    // Cleanup when the channel is closed
+    return () => {
+      wallet.removeListener('state', listener);
+    };
+  });
+
+  try {
+    while (true) {
+      const message = yield take(channel);
+
+      if (message === HathorWallet.ERROR) {
+        yield put(walletStateError());
+        yield cancel();
+      } else {
+        if (wallet.isReady()) {
+          yield put(walletStateReady());
+          yield cancel();
+        }
+
+        continue;
+      }
+    }
+  } finally {
+    if (yield cancelled()) {
+      // When we close the channel, it will remove the event listener
+      channel.close();
+    }
+  }
+}
+
+export function* handleTx(action) {
+  const tx = action.payload;
+  const wallet = yield select((state) => state.wallet);
+
+  if (!wallet.isReady()) {
+    return;
+  }
+
+  // find tokens affected by the transaction
+  const balances = yield call(wallet.getTxBalance.bind(wallet), tx);
+  const stateTokens = yield select((state) => state.tokens);
+  const registeredTokens = stateTokens.map((token) => token.uid);
+
+  let message = '';
+  if (helpers.isBlock(tx)) {
+    message = 'You\'ve found a new block! Click to open it.';
+  } else {
+    message = 'You\'ve received a new transaction! Click to open it.'
+  }
+
+  const notification = walletUtils.sendNotification(message);
+
+  // Set the notification click, in case we have sent one
+  if (notification !== undefined) {
+    notification.onclick = () => {
+      routerHistory.push(`/transaction/${tx.tx_id}/`);
+    }
+  }
+
+  // We should download the **balance** for every token involved in the transaction
+  // and history for hathor
```

Comment:
Why this?

Reply:
It really does not makes sense, fixed in e6a64466cecd5ff1113de007a1e1e74cf5c18290

We should actually download the balance **and history** for all tokens involved in the tx